libminifi/include/core/ProcessContext.h

```diff
@@ -113,10 +113,13 @@ class ProcessContext : public controller::ControllerServiceLookup, public core::
   }
 
   template<typename T>
-  bool getProperty(const std::string &name, T &value) const {
+  std::enable_if_t<!std::is_convertible_v<T&, FlowFile&>, bool> getProperty(const std::string &name, T &value) const {
```

Comment:
I assume this changed to disambiguate it from the newly added overload, but since `Property`  and `std::string` are not interconvertible it is not necessary (even if they were `T` would be resolved to `std::shared_ptr<FlowFile>` and satisfy the non-convertible condition)

Reply:
I think I just generally don't like using `std::string` for a property and didn't want to rely on that type as a discriminator. It compiles with this restriction removed, but I think it's helpful to fail early in this case if someone tries to use a missing overload `(property_name, flow_file) -> optional<T>`.

It also adds complexity to the code, so I can remove it if you think that would be an improvement.

Reply:
Ideally all templates would have proper strict requirements (e.g. in this case convertibility through PropertyValue), but it's tedious to declare all of these until we're able to use C++20 concepts.

Reply:
I just realized the bug you pointed out in the parentheses. Should I change it to exclude shared_ptr<FlowFile>, too, or just remove the condition?

Reply:
I think that the added requirement on this method has merit, but currently nothing verifies it, IMO a test that checks that `ctx.getProperty("blah", flowfile)` does not compile (or something along these lines) would help cement its purpose

Reply:
Added a test in 04ef808